core/app/[locale]/(default)/account/[tab]/_components/addresses-content/add-address.tsx

```diff
@@ -0,0 +1,251 @@
+'use client';
+
+import { Loader2 as Spinner } from 'lucide-react';
+import { useRouter } from 'next/navigation';
+import { useTranslations } from 'next-intl';
+import { useRef, useState } from 'react';
+import { useFormStatus } from 'react-dom';
+import ReCaptcha from 'react-google-recaptcha';
+
+import {
+  createFieldName,
+  FieldNameToFieldId,
+  FieldWrapper,
+  Picklist,
+  PicklistOrText,
+  Text,
+} from '~/app/[locale]/(default)/login/register-customer/_components/register-customer-form/fields';
+import { Link } from '~/components/link';
+import { Button } from '~/components/ui/button';
+import { Field, Form, FormSubmit } from '~/components/ui/form';
+import { Message } from '~/components/ui/message';
+
+import { addAddress } from '../../_actions/add-address';
+
+import {
+  createCountryChangeHandler,
+  createTextInputValidationHandler,
+} from './address-field-handlers';
+import { NewAddressQueryResponseType } from './customer-new-address';
+
+interface FormStatus {
+  status: 'success' | 'error';
+  message: string;
+}
+
+export type AddressFields = NonNullable<
+  NewAddressQueryResponseType['site']['settings']
+>['formFields']['shippingAddress'];
+
+export type Countries = NonNullable<NewAddressQueryResponseType['geography']['countries']>;
+type CountryCode = Countries[number]['code'];
+type CountryStates = Countries[number]['statesOrProvinces'];
+
+interface SumbitMessages {
+  messages: {
+    submit: string;
+    submitting: string;
+  };
+}
+
+const SubmitButton = ({ messages }: SumbitMessages) => {
+  const { pending } = useFormStatus();
+
+  return (
+    <Button
```

Comment:
We have updated the Button component to have a `loading` state. Should use that instead.